rust/numaflow-core/src/pipeline/isb/jetstream/reader.rs

```diff
@@ -385,8 +385,14 @@ impl<C: NumaflowTypeConfig> JetStreamReader<C> {
             .map_err(|e| Error::ISB(format!("Failed to acquire semaphore permit: {e}")))?;
 
         // Apply rate limiting if configured.
-        let effective_batch_size = match &self.rate_limiter {
+        let (effective_batch_size, token_grabbed_epoch) = match &self.rate_limiter {
             Some(rate_limiter) => {
+                // Note the epoch at which we got the estimated batch size from the rate limiter
+                let cur_epoch = std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .expect("Time went backwards beyond unix epoch")
+                    .as_secs();
+
```

Comment:
This may not be accurate, because the cur epoch inside `acquire_n` could be different.